### ISSUE TITLE
ParamSpec misidentified on bound Callable

### ISSUE THREAD

**Issue by second-ed** [astral-sh/ty#3197]
### Summary

ty not identifying when the first arg of a Callable is ParamSpec when within a bound TypeVar

> The first argument to `Callable` must be either a list of types, ParamSpec, Concatenate, or `...`

### `main.py`

```py
from collections.abc import Callable
from typing import ParamSpec, TypeVar

P = ParamSpec("P")
C = TypeVar("C", bound=Callable[P, object])
```

causes ty to complain:

```
The first argument to `Callable` must be either a list of types, ParamSpec, Concatenate, or `...` (invalid-type-form)
```

**Comment by AlexWaygood** [comment]
Thanks for the report! I think ty is correct to issue a diagnostic here, since `P` is unbound in this context. However, the diagnostic message is not great right now. It would be better for us to say "`P` is unbound" or "TypeVar bound cannot be generic" rather than "The first argument to `Callable` must be either a list of types, ParamSpec, Concatenate, or `...`".

**Comment by ratijas** [comment]
Not sure if this is the same underlying issue, but looks similar enough, so I'll throw in my five cents.

This wrapper makes `ty` unhappy, because it thinks the `task` is somehow a coroutine of Unknown return type, rather than R:

```py
import asyncio
import functools
import typing as t

P = t.ParamSpec("P")
R = t.TypeVar("R")


def make_sync(func: t.Callable[P, t.Awaitable[R]]) -> t.Callable[P, R]:
    """
    Wrap an async function so it can be called synchronously.

    See: https://github.com/pallets/click/issues/2033
    """

    @functools.wraps(func)
    def wrapper(*args: P.args, **kwargs: P.kwargs) -> R:
        task = func(*args, **kwargs)
        return asyncio.run(task)

    return wrapper
```

```
Argument to function `run` is incorrect: Expected `Coroutine[Any, Any, Unknown]`, found `Awaitable[R@make_sync]`ty[invalid-argument-type](https://ty.dev/rules#invalid-argument-type)
```

Other than that, VS Code plugin Pylance, when hovered, correctly reports `task` in a popup as `(variable) task: Awaitable[R@make_sync]`

**Comment by dhruvmanila** [comment]
@ratijas I think your issue is different but it seems more an issue in `typeshed` instead because with 3.14 the docs says that both `asyncio.run` and `asyncio.Runner.run` can accept any awaitable object (https://docs.python.org/3.14/library/asyncio-runner.html#asyncio.run) but the PR (https://github.com/python/typeshed/pull/15046) only updated `asyncio.Runner.run`. I think we need to update `typeshed` here.

Edit: I've opened https://github.com/python/typeshed/pull/15807
